src/parser/core/modules/Downtime.js

```diff
@@ -20,19 +20,35 @@ export default class Downtime extends Module {
 		}
 
 		// Merge the downtimes that overlap
-		const finalDowntime = [downtimePeriods.shift()]
+		const finalDowntimes = [downtimePeriods.shift()]
 		downtimePeriods.forEach(dt => {
-			const last = finalDowntime[finalDowntime.length - 1]
+			const last = finalDowntimes[finalDowntimes.length - 1]
 			if (dt.start <= last.end) {
 				if (dt.end > last.end) {
 					last.end = dt.end
 				}
 			} else {
-				finalDowntime.push(dt)
+				finalDowntimes.push(dt)
 			}
 		})
 
+		return finalDowntimes
```

Comment:
Don't forget the early exit return in this function is still of type `number`, which would create errors in the depending functions. Make sure this function always returns Arrays.